travis failure
``
[nomemoryjava] /home/travis/build/openmicroscopy/openmicroscopy/components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java:7347: error: unreported exception Exception; must be caught or declared to be thrown
[nomemoryjava] 	    gw.closeConnector(ctx);
``

--- a/components/blitz/src/omero/gateway/Connector.java
+++ b/components/blitz/src/omero/gateway/Connector.java
@@ -32,6 +32,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.atomic.AtomicLong;
 
 import org.apache.commons.collections.CollectionUtils;
@@ -151,6 +152,7 @@ class Connector
      * instances are stored separately */
     private final Multimap<String, StatefulServiceInterfacePrx> statefulServices;
 
+    /** Map of importStores to prevent re-lookup */
     private final Map<OMEROMetadataStoreClient, String> importStores;
 
     /** Collection of services to keep alive. */
@@ -183,12 +185,11 @@ class Connector
      * Creates a new instance.
      *
      * @param context The context hosting information about the user.
-     * @param secureClient The entry point to server.
+     * @param client The entry point to server.
      * @param entryEncrypted The entry point to access the various services.
      * @param encrypted Pass <code>false</code> to use an unencrypted connection
      *                  for data transfers
      * @param logger Reference to the logger.
-     * @param elapseTime The time between network check.
      * @throws Exception Thrown if entry points cannot be initialized.
      */
     Connector(SecurityContext context, client client,
@@ -203,13 +204,12 @@ class Connector
      * Creates a new instance.
      *
      * @param context The context hosting information about the user.
-     * @param secureClient The entry point to server.
+     * @param client The entry point to server.
      * @param entryEncrypted The entry point to access the various services.
      * @param encrypted Pass <code>false</code> to use an unencrypted connection
      *                  for data transfers
      * @param username The username if this is a derived connector
      * @param logger Reference to the logger.
-     * @param elapseTime The time between network check.
      * @throws Exception Thrown if entry points cannot be initialized.
      */
     Connector(SecurityContext context, client client,
@@ -279,7 +279,8 @@ class Connector
      * Returns the {@link SharedResourcesPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     SharedResourcesPrx getSharedResources()
             throws DSOutOfServiceException
@@ -292,7 +293,8 @@ class Connector
      * Returns the {@link IRenderingSettingsPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IRenderingSettingsPrx getRenderingSettingsService()
             throws DSOutOfServiceException
@@ -306,7 +308,8 @@ class Connector
      * Returns the {@link IRepositoryInfoPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IRepositoryInfoPrx getRepositoryService()
             throws DSOutOfServiceException
@@ -320,7 +323,8 @@ class Connector
      * Returns the {@link IScriptPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IScriptPrx getScriptService()
             throws DSOutOfServiceException
@@ -334,7 +338,8 @@ class Connector
      * Returns the {@link IContainerPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IContainerPrx getPojosService()
             throws DSOutOfServiceException
@@ -348,7 +353,8 @@ class Connector
      * Returns the {@link IQueryPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IQueryPrx getQueryService()
             throws DSOutOfServiceException
@@ -362,7 +368,8 @@ class Connector
      * Returns the {@link IUpdatePrx} service.
      *  
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IUpdatePrx getUpdateService()
             throws DSOutOfServiceException
@@ -376,7 +383,8 @@ class Connector
      * Returns the {@link IMetadataPrx} service.
      *  
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
     IMetadataPrx getMetadataService()
             throws DSOutOfServiceException
@@ -390,7 +398,8 @@ class Connector
      * Returns the {@link IRoiPrx} service.
      *  
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IRoiPrx getROIService()
             throws DSOutOfServiceException
@@ -403,7 +412,8 @@ class Connector
      * Returns the {@link IConfigPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     *@throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IConfigPrx getConfigService()
             throws DSOutOfServiceException
@@ -417,7 +427,8 @@ class Connector
      * Returns the {@link ThumbnailStorePrx} service.
      *
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      ThumbnailStorePrx getThumbnailService()
             throws DSOutOfServiceException
@@ -431,7 +442,8 @@ class Connector
      * Returns the {@link ExporterPrx} service.
      *   
      * @return See above.
-     * @throws @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      ExporterPrx getExporterService()
             throws DSOutOfServiceException
@@ -445,7 +457,8 @@ class Connector
      * Returns the {@link RawFileStorePrx} service.
      *  
      * @return See above.
-     * @throws @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      RawFileStorePrx getRawFileService()
             throws DSOutOfServiceException
@@ -459,7 +472,8 @@ class Connector
      * Returns the {@link RawPixelsStorePrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      RawPixelsStorePrx getPixelsStore()
             throws DSOutOfServiceException
@@ -473,7 +487,8 @@ class Connector
      * Returns the {@link IPixelsPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IPixelsPrx getPixelsService()
             throws DSOutOfServiceException
@@ -487,7 +502,8 @@ class Connector
       * Returns the {@link ITypesPrx} service.
       * 
       * @return See above.
-      * @throws Throwable Thrown if the service cannot be initialized.
+      * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
       */
       ITypesPrx getTypesService()
              throws DSOutOfServiceException
@@ -501,7 +517,8 @@ class Connector
      * Returns the {@link SearchPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      SearchPrx getSearchService()
             throws DSOutOfServiceException
@@ -514,7 +531,8 @@ class Connector
      * Returns the {@link IProjectionPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IProjectionPrx getProjectionService()
             throws DSOutOfServiceException
@@ -528,7 +546,8 @@ class Connector
      * Returns the {@link IAdminPrx} service.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IAdminPrx getAdminService()
             throws DSOutOfServiceException
@@ -543,7 +562,8 @@ class Connector
      * @param secure Pass <code>true</code> to have a secure admin service,
      *               <code>false</code> otherwise.
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      IAdminPrx getAdminService(boolean secure)
             throws DSOutOfServiceException
@@ -560,7 +580,8 @@ class Connector
      * Creates or recycles the import store.
      * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
      */
      OMEROMetadataStoreClient getImportStore()
             throws DSOutOfServiceException
@@ -584,23 +605,33 @@ class Connector
     /**
      * Returns the {@link RenderingEnginePrx Rendering service}.
      * 
+     * @param pixelsID
+     *            The pixels id
+     * @param compression
+     *            A percentage compression level from 1.00 (100%) to 0.01 (1%)
+     *            (the default is 85%)
+     * 
      * @return See above.
-     * @throws Throwable Thrown if the service cannot be initialized.
+     * @throws DSOutOfServiceException
+     *             Thrown if the service cannot be initialized.
+     * @throws ServerError
+     *             Thrown if the service cannot be initialized.
      */
-     RenderingEnginePrx getRenderingService(long pixelsID, float compression)
-            throws DSOutOfServiceException, ServerError
-    {
+    RenderingEnginePrx getRenderingService(long pixelsID, float compression)
+            throws DSOutOfServiceException, ServerError {
         RenderingEnginePrx prx = null;
-        
+
         try {
             if (entryUnencrypted != null) {
                 prx = entryUnencrypted.createRenderingEngine();
             } else {
                 prx = entryEncrypted.createRenderingEngine();
             }
-            this.pcs.firePropertyChange(Gateway.PROP_RENDERINGENGINE_CREATED, null, prx);
+            this.pcs.firePropertyChange(Gateway.PROP_RENDERINGENGINE_CREATED,
+                    null, prx);
         } catch (Exception e) {
-            throw new DSOutOfServiceException("Could not get rendering engine", e);
+            throw new DSOutOfServiceException("Could not get rendering engine",
+                    e);
         }
 
         prx.setCompressionLevel(compression);
@@ -652,7 +683,6 @@ class Connector
      * <code>false</code> otherwise.
      */
      void close(boolean networkup)
-            throws Throwable
     {
         secureClient.setFastShutdown(!networkup);
         if (unsecureClient != null) 
@@ -698,7 +728,6 @@ class Connector
      * <code>false</code> otherwise.
      */
      void closeDerived(boolean networkup)
-            throws Throwable
     {
         for (final Connector c : derived.asMap().values()) {
             try {
@@ -855,9 +884,10 @@ class Connector
      * @param commands The commands to execute.
      * @param target The target context is any.
      * @return See above.
+     * @throws ServerError Thrown if command submission failed
      */
-     CmdCallbackI submit(List<Request> commands, SecurityContext target)
-            throws Throwable
+     CmdCallbackI submit(List<Request> commands, SecurityContext target) throws ServerError
+            
     {
         if (CollectionUtils.isEmpty(commands)) return null;
         DoAll all = new DoAll();
@@ -899,9 +929,10 @@ class Connector
      * @param userName
      *            The name of the user. To be replaced by user's id.
      * @return See above.
+     * @throws ExecutionException Thrown if the connector can't be retrieved.
      */
-     Connector getConnector(final String userName)
-            throws Throwable
+     Connector getConnector(final String userName) throws ExecutionException
+            
     {
         if (StringUtils.isBlank(userName)) 
             return this;

--- a/components/blitz/src/omero/gateway/Gateway.java
+++ b/components/blitz/src/omero/gateway/Gateway.java
@@ -1499,19 +1499,17 @@ public class Gateway {
      * Shuts down the connectors created while creating/importing data for other
      * users.
      *
-     * @param ctx The {@link SecurityContext}
-     * @throws Exception
-     *             Thrown if the connector cannot be closed.
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
      */
-    public void shutDownDerivedConnector(SecurityContext ctx) throws Exception {
+    public void shutDownDerivedConnector(SecurityContext ctx)
+            throws DSOutOfServiceException {
         Connector c = getConnector(ctx, true, true);
         if (c == null)
             return;
-        try {
-            c.closeDerived(isNetworkUp(true));
-        } catch (Throwable e) {
-            throw new Exception("Cannot close the derived connectors", e);
-        }
+        c.closeDerived(isNetworkUp(true));
     }
 
     /**

--- a/components/blitz/src/omero/gateway/Gateway.java
+++ b/components/blitz/src/omero/gateway/Gateway.java
@@ -1296,20 +1296,14 @@ public class Gateway {
      * 
      * @param ctx
      *            The {@link SecurityContext}
-     * @throws Exception
-     *             If the connector is cannot be closed.
      */
-    public void closeConnector(SecurityContext ctx) throws Exception {
+    public void closeConnector(SecurityContext ctx) {
         List<Connector> clist = groupConnectorMap.removeAll(ctx.getGroupID());
         if (CollectionUtils.isEmpty(clist))
             return;
 
         for (Connector c : clist) {
-            try {
-                c.close(isNetworkUp(true));
-            } catch (Throwable e) {
-                throw new Exception("Cannot close the connector", e);
-            }
+            c.close(isNetworkUp(true));
         }
     }
 

--- a/components/blitz/src/omero/gateway/SecurityContext.java
+++ b/components/blitz/src/omero/gateway/SecurityContext.java
@@ -46,7 +46,10 @@ public class SecurityContext {
     /** Host/port of the server */
     private ServerInformation serverInformation;
 
-    /** The compression level */
+    /**
+     * A percentage compression level from 1.00 (100%) to 0.01 (1%) (the default
+     * is 85%)
+     */
     private float compression;
 
     /**
@@ -131,7 +134,9 @@ public class SecurityContext {
     }
 
     /**
-     * Get the compression level
+     * Get the compression level (A percentage compression level from 1.00
+     * (100%) to 0.01 (1%) (the default is 85%))
+     * 
      * @return See above.
      */
     public float getCompression() {
@@ -140,7 +145,10 @@ public class SecurityContext {
 
     /**
      * Set the compression level
-     * @param compression The compression ratio
+     * 
+     * @param compression
+     *            A percentage compression level from 1.00 (100%) to 0.01 (1%)
+     *            (the default is 85%)
      */
     public void setCompression(float compression) {
         this.compression = compression;

--- a/components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java
+++ b/components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java
@@ -1,6 +1,6 @@
 /*
  *------------------------------------------------------------------------------
- *  Copyright (C) 2006-2016 University of Dundee. All rights reserved.
+ *  Copyright (C) 2006-2017 University of Dundee. All rights reserved.
  *
  *
  *  This program is free software; you can redistribute it and/or modify
@@ -7341,11 +7341,28 @@ class OMEROGateway
 		return null;
 	}
 
-	void removeGroup(SecurityContext ctx) {
-	    if (ctx == null)
-	        return;
-	    gw.closeConnector(ctx);
-	}
+    /**
+     * Remove the security group.
+     *
+     * @param ctx
+     *            The security context.
+     * @return See above.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    void removeGroup(SecurityContext ctx) throws DSOutOfServiceException,
+            DSAccessException {
+        if (ctx == null)
+            return;
+        try {
+            gw.closeConnector(ctx);
+        } catch (Throwable t) {
+            handleException(t, "Cannot close connector");
+        }
+    }
 
 	/**
 	 * Loads the file set corresponding to the specified image.

